packages/dialog/src/lit/renderer-directives.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+/* eslint-disable max-classes-per-file */
+import { TemplateResult } from 'lit';
+import { DirectiveResult } from 'lit/directive.js';
+import { LitRendererDirective } from '@vaadin/lit-renderer';
+import { Dialog } from '../vaadin-dialog.js';
+
+export type DialogLitRenderer = (dialog: Dialog) => TemplateResult;
+
+declare abstract class AbstractDialogRendererDirective extends LitRendererDirective<Dialog, DialogLitRenderer> {
+  /**
+   * A property to that the renderer callback will be assigned.
+   */
+  abstract rendererProperty: string;
+
+  /**
+   * Adds the renderer callback to the dialog.
+   */
+  addRenderer(): void;
+
+  /**
+   * Runs the renderer callback on the dialog.
+   */
+  runRenderer(): void;
+
+  /**
+   * Removes the renderer callback from the dialog.
+   */
+  removeRenderer(): void;
+}
+
+export class DialogRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
+
+export class DialogHeaderRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
+
+export class DialogFooterRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty: string;
+}
+
+/**
+ * A Lit directive for populating the content of the dialog.
+ *
+ * ```js
+ * `<vaadin-dialog
+ *   ${dialogRenderer((dialog) => html`...`)}
+ * ></vaadin-dialog>`
+ * ```
+ */
+export declare function dialogRenderer(
+  renderer: DialogLitRenderer,
+  value?: unknown,
```

Comment:
What is `value` in this context? I spend several minutes on this and it wasn't obvious to me. Looking at the code I see that a change in the value triggers a rerender. Some suggestions:
- Maybe call it `dependencies` instead, similar to React hooks? `value` is misleading as it suggests that I have to pass something that is actually used in the rendering
- Add documentation that the parameter exist, what the parameter does, when to use it. In general the docs could be improved to explain the concept of how the directive works.

Reply:
Renamed `value` to `dependencies`. Improved the documentation.

Reply:
Please, let me know if there is anything I could polish in terms of wording and I'll update the Grid PR accordingly.

Reply:
Looks really good now 👍. Added a small suggestion below.